### PR TITLE
Pd137 va abs corr cache 167

### LINKED ISSUE #30578: Adjust the UI of diffraction_adv_setup to include more cache dirs
The instrument team would like to have multiple caching dir candidates when performing reduction for powder diffraction data.
The objective here is to add two additional dir in the UI so that users can pick two additional locations where long term cache files are stored.

> this issue is mirroring this [task](https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/167).

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

This PR includes a UI change where two additional cache dirs are added so that users can have (at most) two additional locations specified where long term cache are stored.

> twin PR: #30580 

**To test:**

After building Mantidworkbench locally, go to the menu bar and follow `Interfaces` -> `Diffraction` -> `Powder Diffraction Reduction`, then select the `Advanced Setup` panel.
To check the responsiveness of the three cache dir fields, put something in the box or use the `Browse` button to select a directory, then save the config into a xml file (use the Save button in the lower right corner).
Open the xml file with any text editor and you should be able to see the three cache directories are concatenated into one single string with ";" in the node `<cachedir>LONG_CACHE_DIR_STRING</cachedir>`.

image

Fixes #30578 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by stonecoldhughes — DISMISSED** [review]
This PR introduces changes to update the UI to accomodate new caching capabilities. No new design changes introduced. Looks fine.

File: scripts/Interface/reduction_gui/reduction/diffraction/diffraction_adv_setup_script.py [hunk 1]
```diff
@@ -277,34 +285,42 @@ def from_xml(self, xml_str):
...
+            'bkgdsmoothpars',
+            'stripvanadiumpeaks',
+            'vanadiumfwhm',
+            'vanadiumpeaktol',
+            'vanadiumsmoothparams',
+            'preserveevents',
+            'extension',
+            'outputfileprefix',
+            # Caching options
+            'cache_dir1',
```
Comment (jmborr) [inline scripts/Interface/reduction_gui/reduction/diffraction/diffraction_adv_setup_script.py:320]: can these be more descriptive? Something like
```python
cache_dir_scan_save
cache_dir_scan_1
cache_dir_scan_2
```

**Review by jmborr — COMMENTED** [review]
I ran the manual test. It produced the XML file as intended

File: scripts/Interface/reduction_gui/reduction/diffraction/diffraction_adv_setup_script.py (hunk 1 above)
Comment (KedoKudo) [reply scripts/Interface/reduction_gui/reduction/diffraction/diffraction_adv_setup_script.py:320]: sure, I will update the name in the next batch of commits.

**Review by jmborr — APPROVED** [review]
:shipit:

**Pull request merged**